src/OpenSage.Mods.Generals/Gui/GeneralsControlBar.cs

```diff
@@ -0,0 +1,285 @@
+
+            _resizeUpBackground = LoadImage(_scheme.ToggleButtonUpOn);
+            _resizeUpHover = LoadImage(_scheme.ToggleButtonUpIn);
+            _resizeUpPushed = LoadImage(_scheme.ToggleButtonUpPushed);
+
+            UpdateResizeButtonStyle();
+
+            State = ControlBarState.Default;
+        }
+
+        // TODO: This should be called at every logic tick.
+        // TODO: This takes a player as the state information. Do we need any other state?
+        public void Update(Player player)
+        {
+            _moneyDisplay.Text = $"$ {player.Money}";
+            State.Update(player, this);
+        }
+
+        public void ToggleSize()
+        {
+            if (_size == ControlBarSize.Maximized)
+            {
+                _window.Top += MinimizeOffset;
+                _background.Top += MinimizeOffset;
+                _size = ControlBarSize.Minimized;
+            }
+            else
+            {
+                _window.Top -= MinimizeOffset;
+                _background.Top -= MinimizeOffset;
+                _size = ControlBarSize.Maximized;
+            }
+
+            UpdateResizeButtonStyle();
+        }
+
+        private void UpdateResizeButtonStyle()
+        {
+            if (_size == ControlBarSize.Maximized)
+            {
+                _resize.BackgroundImage = _resizeDownBackground;
+                _resize.HoverBackgroundImage = _resizeDownHover;
+                _resize.PushedBackgroundImage = _resizeDownPushed;
+            }
+            else
+            {
+                _resize.BackgroundImage = _resizeUpBackground;
+                _resize.HoverBackgroundImage = _resizeUpHover;
+                _resize.PushedBackgroundImage = _resizeUpPushed;
+            }
+        }
+
+        public void AddToScene(Scene2D scene2D)
+        {
+            scene2D.WndWindowManager.PushWindow(_background);
+            scene2D.WndWindowManager.PushWindow(_window);
+        }
+
+        private abstract class ControlBarState
+        {
+            public abstract void OnEnterState(GeneralsControlBar controlBar);
+            public abstract void Update(Player player, GeneralsControlBar controlBar);
+
+            public static ControlBarState Default { get; } = new DefaultControlBarState();
+        }
+
+        private sealed class DefaultControlBarState : ControlBarState
+        {
+            public override void OnEnterState(GeneralsControlBar controlBar)
+            {
+                foreach (var control in controlBar._center.Controls)
+                {
+                    control.Hide();
+                }
+
+                foreach (var control in controlBar._right.Controls)
+                {
+                    control.Hide();
+                }
+            }
+
+            public override void Update(Player player, GeneralsControlBar controlBar)
+            {
+
+            }
+        }
+
+        private sealed class SelectedControlBarState : ControlBarState
```

Comment:
This is probably not the final design, since all control bar states (except the default) are somehow related to selection. Those details need to be ironed out, but that can wait for the next PR.